minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-framework-core/src/main/java/org/apache/nifi/minifi/c2/C2NifiClientService.java

```diff
@@ -367,4 +368,27 @@ private List<ProcessorBulletin> getBulletins(int processorBulletinLimit) {
         }
         return new ArrayList<>();
     }
+
+    private List<ProcessorStatus> getProcessorStatus() {
+        return flowController.getEventAccess()
+                .getGroupStatus(ROOT_GROUP_ID)
+                .getProcessorStatus()
+                .stream()
+                .map(status -> {
+                    ProcessorStatus processorStatus = new ProcessorStatus();
```

Comment:
Very minor, but maybe we could extract this to a separate method to make the streaming pipeline more concise

Reply:
I followed the patter what was used for 'queues'  here as well but for sure I can extract the initiation of ProcessorStatus into a separate method.

Reply:
Done